### PR TITLE
Remove GenControlBlock code

### PR DESCRIPTION
**Pull request by amankumar1**
The flag _isGenControlBlock was used in a project that existed
prior to this code being contributed and is not relevant to the
open source code base.

Signed-off-by: Aman Kumar <amank@ca.ibm.com>

### CODE REVIEW

File: compiler/optimizer/OrderBlocks.hpp [hunk 1]
```diff
@@ -121,7 +121,6 @@ class TR_OrderBlocks : public TR_BlockOrderingOptimization
    void            peepHoleBranchBlock(TR::CFG *cfg, TR::Block *block, char *title);
    void            peepHoleBranchAroundSingleGoto(TR::CFG *cfg, TR::Block *block, char *title);
    bool            peepHoleBranchToFollowing(TR::CFG *cfg, TR::Block *block, TR::Block *followingBlock, char *title);
-   bool            peepHoleBranchAroundSimpleGenControlBlocks(TR::CFG *cfg, TR::Block *block, TR::Block *followingBlock, char *title);
```
Comment (amankumar1) [inline compiler/optimizer/OrderBlocks.hpp]: Not sure about this function. Folded away as `false`.

**Comment by vijaysun-omr** [comment]
Are the following flags also in the same category that can be removed 

       _isGenAsmBlock                          = 0x00010000,
       _isGenAsmFlowBlock                    = 0x00100000,

File: compiler/optimizer/OrderBlocks.hpp (hunk 1 above)
Comment (mgaudet) [inline compiler/optimizer/OrderBlocks.hpp]: I think that's the right call.

**Comment by vijaysun-omr** [comment]
I am not going to hold this up until the other flags are removed. There should be separate commit coming at some point doing that; @amankumar1 do you plan to do that cleanup soon too ?

**Comment by amankumar1** [comment]
@vijaysun-omr I have cleaned up the other flags as well. See #440.

**Pull request merged**